### ISSUE TITLE
Travis uses treats Objective-C files as C++

### ISSUE THREAD

**Issue by billyquith** [andlabs/libui#108]
Try not specifying the languages in the `project()` declaration. You don't include Obj-C so it may have been disabled. They will be auto detected if you don't specify them.

This works with cmake 3.5 and Ninja on OSX. I.e. without this change.

**Comment by kainjow** [comment]
You could try `set_source_files_properties` and set LANGUAGE to C. Or remove `-std=c++11` as there aren't any C++ files except Windows.

**Comment by andlabs** [comment]
Not specifying the languages for `project()` does not work: https://github.com/andlabs/libui/pull/114 still fails

Will try the other things after merging that PR.

**Comment by kainjow** [comment]
You can get old 2.8 versions here: https://cmake.org/files/v2.8/ - `target_include_directories` and `target_compile_options` and `target_compile_definitions` require 3.0. These, along with `target_sources` (requires 3.1) are what make modern CMake much easier, whereas the 2.x versions require nasty global functions that apply to all targets (`add_definitions` and `include_directories`).

The only "problem" with requiring CMake 3.0+ is *nix platforms which usually only have 2.8 in their repositories. However Ubuntu 16.04 has 3.5 and Debian 8 has 3.0.

I would suggest considering requiring 3.0 and drop 2.8 support.

**Comment by andlabs** [comment]
I'd like to do that too; it'd make following the documentation a lot easier, since it sure as hell doesn't say what the minimum version of everything is! But then I'm afraid of using a 3.1-3.5 feature by accident... Surely there has to be SOME warning (FATAL_ERROR?).

**Comment by andlabs** [comment]
```
[09:49:44]  <+ngladitz>	then you probably hit https://cmake.org/gitweb?p=cmake.git;a=commit;h=95cdf132489c79e88a10fdf7a7566fa002c7680b
[09:50:16]  <+ngladitz>	the fix in that commit is in cmake since v3.1.0
```
I'll probably just bump the minimum required cmake version to that.
